volatility3/framework/automagic/symbol_cache.py

```diff
@@ -22,164 +25,390 @@
+    def __init__(self, filename: str):
+        super().__init__()
+        self._filename = filename
+        self._classifiers = {}
+        for subclazz in volatility3.framework.class_subclasses(IdentifierProcessor):
+            self._classifiers[subclazz.operating_system] = subclazz
+
+    def add_identifier(self, location: str, operating_system: str, identifier: str):
+        """Adds an identifier to the store"""
+        pass
+
+    def find_location(self, identifier: bytes, operating_system: Optional[str]) -> Optional[str]:
+        """Returns the location of the symbol file given the identifier
+
+        Args:
+            identifier: string that uniquely identifies a particular symbol table
+            operating_system: optional string to restrict identifiers to just those for a particular operating system
+
+        Returns:
+            The location of the symbols file that matches the identifier
+        """
+        pass
+
+    def get_local_locations(self) -> List[str]:
+        """Returns a list of all the local locations"""
+        pass
+
+    def update(self):
+        """Locates all files under the symbol directories.  Updates the cache with additions, modifications and removals.
+        This also updates remote locations based on a cache timeout.
+
+        """
+        pass
+
+    def get_identifier_dictionary(self, operating_system: Optional[str] = None, local_only: bool = False) -> \
+            Dict[bytes, str]:
+        """Returns a dictionary of identifiers and locations
+
+        Args:
+            operating_system: If set, limits responses to a specific operating system
+            local_only: Returns only local locations
+
+        Returns:
+            A dictionary of identifiers mapped to a location
+        """
+        pass
+
+    def get_identifier(self, location: str) -> Optional[bytes]:
+        """Returns an identifier based on a specific location or None"""
+        pass
+
+    def get_identifiers(self, operating_system: Optional[str]):
+        """Returns all identifiers for a particular operating system"""
+        pass
+
+    def get_location_statistics(self, location: str) -> Optional[Tuple[int, int, int, int]]:
+        """Returns ISF statistics based on the location
+
+        Returns:
+            A tuple of base_types, types, enums, symbols, or None is location not found"""
+
+    def get_hash(self, location: str) -> bool:
+        """Returns the hash of the JSON from within a location ISF"""
+
+
+class SqliteCache(CacheManagerInterface):
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    cache_period = '-3 days'
+
+    def __init__(self, filename: str):
+        super().__init__(filename)
+        try:
+            self._database = self._connect_storage(filename)
+        except sqlite3.DatabaseError:
+            os.unlink(filename)
+            self._database = self._connect_storage(filename)
+
+    def _connect_storage(self, path: str):
```

Comment:
It seems to be return the `Connection` Object of SQLite3.

```suggestion
    def _connect_storage(self, path: str) -> sqlite3.Connection:
```